lib/proxy/pluggable.rb

```diff
@@ -100,6 +86,7 @@ def default_settings(a_hash = {})
     end
 
     def initialize_after(*module_names)
+      logger.warn("#{plugin_name}: 'initialize_after' has been deprecated and will be removed in future versions.")
```

Comment:
It appears that this isn't actual functional any more, since `validate_dependencies!` is no longer called with `@initialize_after`.  A few points follow:

1. Was there any use for feature this in existing plugins?  It looks like we now lose the ability of a provider or module to order itself after another module, and I don't know if we even take the `require` into account for ordering.  It still seems useful to me, despite the general ordering improvements in this PR.
2. It's not so much deprecated as disabled, so the code below should be removed as I don't think it's functional.
3. If being removed, it would be better to raise an exception as it's going to behave very differently.